9-react.ts+TDD/src/App.test.tsx

```diff
@@ -0,0 +1,85 @@
+import faker from "faker";
+import puppeteer, { ElementHandle, Page, Browser } from "puppeteer";
+import { TodoItemDTO } from "../../common/interfaces";
+
+
+const todoItem: TodoItemDTO = {
+    text: faker.random.words(3),
+    id: faker.datatype.uuid(),
+    isChecked: faker.datatype.boolean(),
+  };
+
+
+const initTestEnv = async (): Promise<Page> => {
+  const browser: Browser = await puppeteer.launch({
+    headless: true,
+    devtools: false,
+  });
+
+  const page: Page = await browser.newPage();
+
+  page.emulate({
+    viewport: {
+      width: 500,
+      height: 900
+    },
+    userAgent: ''
+  });
+
+  await page.goto('http://localhost:8080/');
+  await page.waitForSelector('.appBodyContainer-0-2-3');
+
+  return page;
+}
+
+
+const addNewItem = async (page: Page): Promise<void> => {
+  await page.click(".newItemButton-0-2-8");
+  await page.type("input[name=add-new-item]", todoItem.text);
+  await page.click(".submitButton-0-2-12");
+}
+
+describe('E2E', () => {
+    test('Add new item', async () => {
+      const page: Page = await initTestEnv();
+      await addNewItem(page);
+      await page.waitForSelector('div[data-hook=todo-item]');
+    })
+
+    test('Delete item', async () => {
+      const page = await initTestEnv();
+      await addNewItem(page);
+      await page.waitForSelector('div[data-hook=todo-item]');
+      const actionItems: Array<ElementHandle> = await page.$$('.icon-0-2-14');
```

Comment:
what's this `.icon-0-2-14` it looks very unstable.. as in it's generated code that can change frequently